src/middlewared/middlewared/plugins/enterprise/m_series_nvdimm_firmware_freebsd.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding=utf-8 -*-
+import logging
+import os
+import re
+
+import sysctl
+
+from middlewared.service import CallError, private, Service
+
+logger = logging.getLogger(__name__)
+
+
+class EnterpriseService(Service):
+
+    SIZE = None
+    SIZE_ERROR = "Size not retrieved yet"
+
+    VERSION = None
+    VERSION_ERROR = "Version not retrieved yet"
+
+    @private
+    def setup_m_series_nvdimm(self):
+        try:
+            self.SIZE = int(re.match("NVDIMM region ([0-9]+)GB", sysctl.filter("dev.nvdimm.0.%desc")[0].value).group(1))
```

Comment:
As far as I know nvdimm may not exist, so first we need to make sure the device is there.

Reply:
If it's an `ECHOWARP` platform, then it should have an NVDIMM as they come standard with the m-series.